### PR TITLE
insights: resolve next insight recording as the first of the next month

### LINKED ISSUE #23767: insights: use consistent time intervals for indexed recordings
Currently, indexed recordings operate by executing shortly after an insight is created, and are then delayed by a configurable number of days. This leads to strange behavior when crossing from historical to non-historical data, as the time interval and frequency changes drastically. This issue is to align the indexed recordings to always take place at the first of the next month after the insight is created, or after a prior recording. 

https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/enterprise/internal/insights/store/insight_store.go?utm_product_name=GoLand&utm_product_version=GoLand#L256:28

### PR DESCRIPTION
**Pull request by coury-clark**
Closes #23767
We are changing the recording scheduler to use a fixed 1st of every month.

### CODE REVIEW

**Review by camdencheek — APPROVED** [review]

File: internal/insights/insights_test.go [hunk 1]
```diff
@@ -174,3 +176,25 @@ const insightSettingSimple = `{"searchInsights.insight.global.simple": {
       "weeks": 2
     }
   }}`
+
+func TestNextRecording(t *testing.T) {
+	type args struct {
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply internal/insights/insights_test.go:181]: Interestingly, this is how goland autogenerates a table test.

File: internal/insights/insights.go [hunk 2]
```diff
@@ -256,3 +257,9 @@ const (
 	User SettingFilter = "user"
 	All  SettingFilter = "all"
 )
+
+// NextRecording calculates the time that a series recording should occur given the current or most recent recording time.
+func NextRecording(current time.Time) time.Time {
+	year, month, _ := current.Date()
```
Comment (camdencheek) [inline internal/insights/insights.go:263]: I _think_ we're using `time.UTC` everywhere, but shouldn't this inherit the timezone from `current`? 

Or, if we want to the return value to always be in UTC, we should probably convert `current` to `UTC` first. 

```suggestion
	year, month, _ := current.In(time.UTC).Date()
```

File: internal/insights/insights_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply internal/insights/insights_test.go:181]: Huh, weird 🤷

**Review by coury-clark — COMMENTED** [review]
It seems if I press the review button on accident, there is no way to escape without a comment.

**Pull request merged**